ambry-router/src/main/java/com.github.ambry.router/GetOperation.java

```diff
@@ -172,9 +153,9 @@ void processServerError(ServerErrorCode errorCode) {
    * @param exception the {@link RouterException} to possibly set.
    */
   void setOperationException(RouterException exception) {
-    if (operationException == null || exception.getErrorCode() == RouterErrorCode.BlobDeleted
+    if (operationException.get() == null || exception.getErrorCode() == RouterErrorCode.BlobDeleted
```

Comment:
May I know why we over-ride if operationException is not set(i.e, if its set to null). Also, can you also explain what happens when all replicas throws InternalServerError for some reason, what will be the reconciled error code we get ?